app/models/chargeback_vm.rb

```diff
@@ -0,0 +1,185 @@
+    :fixed_storage_2_cost     => :float,
+    :fixed_2_cost             => :float,
+    :fixed_cost               => :float,
+    :memory_allocated_cost    => :float,
+    :memory_allocated_metric  => :float,
+    :memory_used_cost         => :float,
+    :memory_used_metric       => :float,
+    :memory_cost              => :float,
+    :memory_metric            => :float,
+    :net_io_used_cost         => :float,
+    :net_io_used_metric       => :float,
+    :net_io_cost              => :float,
+    :net_io_metric            => :float,
+    :storage_allocated_cost   => :float,
+    :storage_allocated_metric => :float,
+    :storage_used_cost        => :float,
+    :storage_used_metric      => :float,
+    :storage_cost             => :float,
+    :storage_metric           => :float,
+    :total_cost               => :float
+  )
+
+  def self.build_results_for_report_ChargebackVm(options)
+    # Options:
+    #   :rpt_type => chargeback
+    #   :interval => daily | weekly | monthly
+    #   :start_time
+    #   :end_time
+    #   :end_interval_offset
+    #   :interval_size
+    #   :owner => <userid>
+    #   :tag => /managed/environment/prod (Mutually exclusive with :user)
+    #   :chargeback_type => detail | summary
+    _log.info("Calculating chargeback costs...")
+
+    tz = Metric::Helper.get_time_zone(options[:ext_options])
+    # TODO: Support time profiles via options[:ext_options][:time_profile]
+
+    interval = options[:interval] || "daily"
+    cb = new
+    report_user = User.find_by(:userid => options[:userid])
+
+    # Find Vms by user or by tag
+    if options[:owner]
+      user = User.find_by_userid(options[:owner])
+      if user.nil?
+        _log.error("Unable to find user '#{options[:owner]}'. Calculating chargeback costs aborted.")
+        raise MiqException::Error, _("Unable to find user '%{name}'") % {:name => options[:owner]}
+      end
+      vms = user.vms
+    elsif options[:tag]
+      vms = Vm.find_tagged_with(:all => options[:tag], :ns => "*")
+      vms &= report_user.accessible_vms if report_user && report_user.self_service?
+    elsif options[:tenant_id]
+      tenant = Tenant.find(options[:tenant_id])
+      if tenant.nil?
+        _log.error("Unable to find tenant '#{options[:tenant_id]}'. Calculating chargeback costs aborted.")
+        raise MiqException::Error, "Unable to find tenant '#{options[:tenant_id]}'"
+      end
+      vms = tenant.vms
+    else
+      raise _("must provide options :owner or :tag")
+    end
+    return [[]] if vms.empty?
+
+    vm_owners = vms.inject({}) { |h, v| h[v.id] = v.evm_owner_name; h }
+    options[:ext_options] ||= {}
+
+    base_rollup = MetricRollup.includes(
+        :resource           => :hardware,
+        :parent_host        => :tags,
+        :parent_ems_cluster => :tags,
+        :parent_storage     => :tags,
+        :parent_ems         => :tags)
+                      .select(*Metric::BASE_COLS).order("resource_id, timestamp")
```

Comment:
@zeari for some reason this `MetricRollup.includes` block got a new/wrong indentation can you revert it to its original one? Thanks.